src/charm.py

```diff
@@ -150,14 +197,18 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
             self.unit.status = BlockedStatus(
                 f"Failed to start timer for regular reconciliation and binary update checks: {ex}"
             )
+        # Safe guard against unexpected error.
+        except Exception as err:  # pylint: disable=broad-exception-caught
```

Comment:
Do you think it might be worth making this a decorator? We seem to repeat this `Exception` catching in a few functions: https://peps.python.org/pep-0318/

Reply:
Based on complexity, this could be a follow up issue, up to you

Reply:
I wrote the decorators.